include/dlaf/types.h

```diff
@@ -46,6 +57,32 @@ enum class Backend {
 #endif
 };
 
+inline std::ostream& operator<<(std::ostream& os, const Backend& backend) {
+  if (backend == Backend::MC) {
+    os << "CPU";
+  }
+  else if (backend == Backend::GPU) {
+    os << "GPU";
+  }
+  return os;
+}
+
+// TODO: Do we need this? Or should we be explicit about which Backend goes with
+// which Device?
+/// Suitable device given a backend.
+template <Backend backend>
+struct BackendDevice;
```

Comment:
A way to get a default device given a backend is a good idea,
however, I would choose another name (e.g. `DefaultDevice<Backend>` or `PreferredDevice<Backend>`, but feel free to come up with other suggestions).

Reply:
Yep, agreed that the name isn't ideal! I think `DefaultDevice` sounds reasonable. For this PR we don't need it, but should we have the inverse as well, i.e. `DefaultBackend<Device>`?

Reply:
I've renamed it to `DefaultDevice` but I'll leave this unresolved to let others comment on the name next week.

Reply:
I'm closing this as resolved based on the silence and :+1:s.